syntax = "proto3";

package arrus.proto;

import "io/proto/common/LinearFunction.proto";

message RxSettings {
  // optional
  oneof dtgcAttenuation_ {
    uint32 dtgc_attenuation = 2;
  }
  uint32 pga_gain = 3;
  uint32 lna_gain = 4;
  LinearFunction tgc_curve_linear = 5;
  repeated double tgc_samples = 6;

  uint32 lpf_cutoff = 7;

  oneof activeTermination_ {
    uint32 active_termination = 9;
  }
}